src/raiden_libs/user_address.py

```diff
@@ -158,13 +158,16 @@ def populate_userids_for_address(self, address: Address, force: bool = False) ->
         if force or not self.get_userids_for_address(address):
             self.add_userids_for_address(
                 address,
-                (
-                    user.user_id
-                    for user in self._client.search_user_directory(to_normalized_address(address))
-                    if self._validate_userid_signature(user)
-                ),
+                (user.user_id for user in self.get_users_from_address(address)),
             )
 
+    def get_users_from_address(self, address: Address) -> List[User]:
```

Comment:
As mentioned above, we should not use search_user_directory at all. That has several reason which have been introduced by the presenceless changes.

1. The Raiden node only asks on demand for a user_id where to send messages to. It expects only one user_id
2. A Raiden node is never intended to be online at two servers simultaneously
3. The PFS should only have the one source of truth, that is the received presence information. The only way where two users could be online simultaneously is if the user roams between servers. Then the PFS should always accept the latest received presence update.
4. Most likely the search user directory call should fail anyways as the users do not share a room with the PFS

Reply:
I'm a bit confused about the expected behaviour. This function came to be as a refactoring of a very nested list comprehension, and also so it could be reused, since the only way I found around the code to access the displayname is to have access to a user object.

All the things around this function suggests that we can expect that one address will point to more than one use, from this function (that used to be a comprehension) to the name of the function where it was extracted from, `populate_userids_for_address`.

So I remain with 2 questions:
1. Can we be really sure that `search_user_directory` will only return one user, despite the evidence?
2. If we shouldn't ever user the user directory, what part of the code is fetching and storing the displayname?

Reply:
As per spec [presence updates](https://matrix.org/docs/spec/client_server/r0.6.0#m-presence) come with the displayname if there is any. We have seen that this is not always but most of the times the case. So with any presence event arriving in the sync response, it should contain the respective displayname of the user. If not and if there isnt any displayname in the displaycache it should probably be fetched right away by the `user.get_display_name()` call. 

`search_user_directory` should never be called and it probably never will return a user for an address as the PFS user does not share a room nor a public room with it.

Nevertheless, the displayname should already arrive with the presence update. The respetctive user object could be created right away.